### ISSUE TITLE
tf.losses.mean_squared_error returns a list in tensorflow '2.0.0-rc1'

### ISSUE THREAD

**Issue by alibeyram** [tensorflow/tensorflow#32911]
Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
OS Platform and Distribution (e.g., Linux Ubuntu 16.04): 
```uname -mrs
Linux 3.10.0-514.el7.x86_64 x86_64
```
TensorFlow installed from (source or binary): binary (pip)
TensorFlow version (use command below): 'v2.0.0-rc0-101-gd2d2566'
Python version: 3.6.6 :: Anaconda, Inc.
Bazel version (if compiling from source): N/A
GCC/Compiler version (if compiling from source): N/A
CUDA/cuDNN version: N/A
GPU model and memory: N/A

I just updated my TensorFlow to version:
```
tf.__version__
'2.0.0-rc1'
```
using pip install.  I am using python version:

```
python -V
Python 3.6.6 :: Anaconda, Inc.
```

the bug is that `tf.losses.mean_squared_error`  returns a list rather than a scaler. A simple code to replicate this:

```
np.random.seed(seed=10)
a, b = np.random.rand(5,1), np.random.rand(5,1)
tf.losses.mean_squared_error(a,b).numpy()
```

returns:

`array([0.29868848, 0.03143916, 0.01609916, 0.33604403, 0.16823713])`

you can find a way around it by using 

`tf.losses.MeanSquaredError()(a,b).numpy()`

which returns:
`0.17010`

**Comment by gowthamkpr** [comment]
Yes the behavior of the same function in [1.15](https://colab.sandbox.google.com/gist/gowthamkpr/d9126a9645e900c67e48c05d4f620590/1-15-mse-loss.ipynb) and [2.0.o.rc2](https://colab.sandbox.google.com/gist/gowthamkpr/d9126a9645e900c67e48c05d4f620590/1-15-mse-loss.ipynb) is not the same.

**Comment by pavithrasv** [comment]
@alibeyram The mean_squared_error function returns one value per example, `MeanSquaredError` class which is the 2.0 way of specifying metrics uses mean_squared_error internally and computes the the weighted mean of the list of the values across all samples.

**Comment by alibeyram** [comment]
@pavithrasv So I think we should not call it `mean_squared_error` since it does not do any averaging and it's just `(y_true - y_pred)**2`

**Comment by pavithrasv** [comment]
It computes the mean across all dimensions other than the batch dimension. If you provide 3D input the output will be 1D.

**Comment by alibeyram** [comment]
@pavithrasv Thank you for clearing it up. The only suggestion I have is that if that's a change form  version 1 to version 2, it is useful to mention it in https://www.tensorflow.org/guide/migrate

**Comment by alibeyram** [comment]
@pavithrasv Also it does not return a 1D when you provide 3D.

```
np.random.seed(seed=10)
a, b = np.random.rand(2,2,1), np.random.rand(2,2,1)
tf.losses.mean_squared_error(a,b).numpy()
```

returns:

```array([[7.44272772e-02, 4.16342380e-02],
       [1.89734615e-01, 1.37518534e-04]])```

which is a 2D array. 

The code only takes mean over the very last dimension.
take a look at the [code](https://github.com/tensorflow/tensorflow/blob/r2.0/tensorflow/python/keras/losses.py#L762-L771) 

`K.mean(math_ops.squared_difference(y_pred, y_true), axis=-1)`

**Comment by pavithrasv** [comment]
Oh sorry, i was wrong, it computes mean along the last dimension. How is the behavior different between 1.15 and 2.0 ? Are you referring to the standalone behavior or behavior in a Keras model fit?

**Comment by alibeyram** [comment]
@pavelbulanov Look at @gowthamkpr comment above. There are two code collab attach that shows one returns list and the other one return a scaler

**Comment by alibeyram** [comment]
@pavithrasv Also for me it does not make sense to compute mean along the last dimension.

**Comment by alibeyram** [comment]
So As you said I think this is the intended behavior which is different between versions [1.14](https://github.com/tensorflow/tensorflow/blob/r1.14/tensorflow/python/keras/losses.py#L100) and [2.0](https://github.com/tensorflow/tensorflow/blob/r2.0/tensorflow/python/keras/losses.py#L100)

**Comment by jvishnuvardhan** [comment]
@alibeyram I am closing this issue as it is intended behavior. If your use case require additional features, then please raise new issue as a `feature request. Thanks

**Comment by pavithrasv** [comment]
In TF 1.x we had different loss functions in `tf.losses` and `tf.keras.losses`, we unified them in TF 2.0. We went the `tf.keras.losses` way for the functions where the inputs are expected to be at least 2D and we mean across the last dimension.
